renku/core/commands/update.py

```diff
@@ -143,7 +146,9 @@ def execute_workflow(
 
     started_at_time = local_now()
 
-    modified_outputs = _execute_workflow_helper(plans=plans, client=client)
+    # NOTE: Create a ``CompositePlan`` because ``workflow_covert`` expects it
+    workflow = CompositePlan(id=CompositePlan.generate_id(), plans=plans, name=f"plan-collection-{uuid.uuid4().hex}")
```

Comment:
i dont like this as there's a `CompositePlan` -> `List[Plan]` -> `CompositePlan` cycle going on here. but for the sake of having the execute feature in master i'd merge this and the toil PR in the meanwhile i'll finish the refactor of handling list of plans in the workflow execute/converter plugins

Reply:
Me and Mohammad discussed this in the past, and it is unfortunate but a bit of a necessary feature to allow the same executor to be used with `workflow execute`, `rerun` and `update`. In the rerun and update case, there is only a list of plans as the source, not a CompositePlan, and we didn't want to modify `execute_workflow` as you're still working on that part.

But we should consider not building a CompositePlan in the `execute_workflow` method as it's not really needed.